providers/implementations/keymgmt/dh_kmgmt.c

```diff
@@ -222,6 +266,33 @@ static int dh_get_key_params(void *key, OSSL_PARAM params[])
     return dh_get_dpk_params(key, params);
 }
 
+static int dh_iskey(const void *key)
+{
+    return DH_get0_pub_key(key) != NULL;
```

Comment:
If we can set keys up via setparams would it be possible to only have the private key bit?

Reply:
The general assumption with openssl is that a valid key pair *always* has the public half. A lot will fall apart if that doesn't hold true.